lib/pmdtester/builders/summary_report_builder.rb

```diff
@@ -0,0 +1,146 @@
+require_relative './html_report_builder'
+require_relative '../pmd_branch_detail'
+
+module PmdTester
+  # Building summary report to show the details about projects and pmd branchs
+  class SummaryReportBuilder < HtmlReportBuilder
+    REPORT_DIR = 'target/reports/diff'.freeze
+    BASE_CONFIG_PATH = 'target/reports/diff/base_config.xml'.freeze
+    PATCH_CONFIG_PATH = 'target/reports/diff/patch_config.xml'.freeze
+    INDEX_PATH = 'target/reports/diff/index.html'.freeze
+
+    def build(projects, base_name, patch_name)
+      @projects = projects
+      @base_details = get_branch_details(base_name)
+      @patch_details = get_branch_details(patch_name)
+
+      FileUtils.mkdir_p(REPORT_DIR) unless File.directory?(REPORT_DIR)
+      index = File.new(INDEX_PATH, 'w')
+
+      html_report = build_html_report('Summary report')
+      copy_css(REPORT_DIR)
```

Comment:
I think, this should be optimized. We copy the css files for the summary report and for each project report.
We should be able to reuse the css files for the project reports from the summary report.

Reply:
Each project has its own css resource file that can make each project's diff html report independent,
which means we can upload single project diff report to somewhere, although, I don't know if there will be such a usage scenario in the future :smile: